src/Dhall.hs

```diff
@@ -203,9 +212,32 @@ inputFromWith
     -- ^ The Dhall program
     -> IO a
     -- ^ The decoded value in Haskell
-inputFromWith filename (Type {..}) ctx n txt = do
+inputFromWith filename ty ctx n txt =
+  inputDirFromWith "." filename ty ctx n txt
+
+{-| Extend 'inputFrom' with a root directory to resolve imports relative
+    to, a file to mention in errors as the source, a custom typing
+    context, and a custom normalization process.
+
+@since 1.6
+-}
+inputDirFromWith
```

Comment:
Are you sure that you want this function to take two `FilePath` arguments.  Couldn't you infer the starting directory from the path of the source file?

Reply:
That's not guaranteed to be a path to the file; e.g., it's also `<input>` or `<stdin>` if that's where the data is actually coming from.

Reply:
Yeah, and also thinking more about your answer in the other issue I agree that it's necessary to disentangle the two concepts, so don't worry about this after all